src/optimize.rs

```diff
@@ -120,6 +123,34 @@ pub fn prep_ilp_data(
             );
             cost_i.push(cost_model.get_self_cost(egraph, node));
             g_i.push(m);
+            use crate::model::Mdl::*;
+            fus_i.push(match node {
+                Input(..) => true,  
+                CompareOp(..) => true,
+                BroadcastInDimOp(..) => true, 
+                ConvertOp(..) => true,
+                ReshapeOp(..) => true,
+                GatherOp(..) => true,
+                ConcatenateOp(..) => true,
+                SliceOp(..) => true, 
+                TransposeOp(..) => true,
+                MulOp(..) => true,
+                AddOp(..) => true,
+                DivOp(..) => true,
+                SubtractOp(..) => true,
+                MinOp(..) => true,
+                MinOp(..) => true,
+                NegOp(..) => true, 
+                TanhOp(..) => true, 
+                ExpOp(..) => true,
+                IotaOp(..) => true, 
+                DynamicUpdateSliceOp(..) => true,
+                DynamicSliceOp(..) => true,
+                ScatterOp(..) => true,
+                ReturnOp(..) => true,
+                BlackBox(..) => false,
+                _ => false,
```

Comment:
nit: annoying but maybe we should write the other cases explicitly so that we get reminded to update this whenever we add a new op

(we are already quite bad at doing this and have a lot of wildcards, but I think most of them will throw an exception so it'll get caught in testing at least - but this particular case seems useful enough to do it)

Reply:
🙂‍↕️

Reply:
![image](https://github.com/user-attachments/assets/b4c58091-bb23-4003-9222-a4c00c48c9d4)